src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -384,6 +377,19 @@ RouterExecutorRun(QueryDesc *queryDesc, ScanDirection direction, long count)
 		(*destination->rStartup)(destination, operation, queryDesc->tupDesc);
 	}
 
+	/* we only support returning nothing or scanning forward */
```

Comment:
You mean to to true? Because otherwise we'd just re-execute the query. That'd then access the tuplestorestate again. Which'd potentially cause problems. So we'd have to also unset the tuplestorestate. Don't think that's going to be simpler.